COM-93 Implement dispensation reports

tmvumbi2 opened the pull request:
Jira ticket: https://jembiprojects.jira.com/browse/COM-93

tmvumbi2 committed:
COM-93 Implement dispensation reports
Implements dispensation reports

lauravignoli commented on openmrs/apps/reports/DISPENSATION/sql/prescribed_vs_dispensed.sql:
```diff
@@ -1,105 +1,105 @@
-    0 AS 'prescribed 35',
-    0 AS 'dispensed 35';
\ No newline at end of file
+SELECT 'ABC/3TC' AS 'Title 1',
```
@tmvumbi2 these drugs don't match the existing drugs in our instance, specifically:

- some don't exist
- existing drug AZT is not present
- existing drug AZT/3TC/NVP shows incorrectly as AZT/3TC+NVP

lauravignoli replied:
Richard will update the template to reflect the correct drug names.

lauravignoli replied:
New template:

https://docs.google.com/document/d/1-FjSK9WNp40zYvWIA-0LxiZqBIAK7ZiCn4tKI95ew_c/edit

tmvumbi2 committed:
COM-93 Refactor dispensation queries

Simplifies the query to calculate the ARV dispensed by category: dynamically retrieves the list of drugs from the ARV concept set and then calculates the total ARV dispensed for each category; all is done in one query. This approach reduces the number of reports sent to Excel from 7 to 1 which improves the rendering speed in Excel.

Simplifies the query prescribed_vs_dispensed : dynamically retrieves the list of drugs from the ARV concept set and then calculates the total prescribed and dispensed for each drug. This change makes the query simpler and cleaner by reducing the number of columns

lauravignoli commented:
@tmvumbi2 The template is zoomed out now which makes it very difficult to see. Can you please update the size?

lauravignoli committed:
COM-93 Zoom 75% in on dispensation template

This was done because it was too small to view.
